ring/lifecycler_test.go

```diff
@@ -98,6 +98,113 @@ func TestLifecycler_HealthyInstancesCount(t *testing.T) {
 	})
 }
 
+func TestLifecycler_InstancesInZoneCount(t *testing.T) {
+	ringStore, closer := consul.NewInMemoryClient(GetCodec(), log.NewNopLogger(), nil)
+	t.Cleanup(func() { assert.NoError(t, closer.Close()) })
+
+	var ringConfig Config
+	flagext.DefaultValues(&ringConfig)
+	ringConfig.KVStore.Mock = ringStore
+
+	instances := []struct {
+		zone                                string
+		healthy                             bool
+		expectedHealthyInstancesInZoneCount int
+		expectedHealthyInstancesCount       int
```

Comment:
It's the number of expected zones, not healthy instances count.

```suggestion
		expectedZonesCount       int
```